status-im/status-react
# [#3174] add js injection support for service worker in DApp

fixes #3174 

### Summary:

Add support for JS injection when DApp using Service Worker API on Android.

### Steps to test:
- Open Status
- Open DApp https://s3-us-west-1.amazonaws.com/turnsy/dapp/www/index.html
- Verify that web3 object is loaded and accessible
- Reload DApp (Go back to chat list and open it again)
- Check that web3 is still loaded and accessible

### Review Notes

Actual changes are here https://github.com/status-im/react-native-webview-bridge/compare/feature/camera-permissions...status-im:fix-web3-injection

### Testing Notes

Android only (versions 7 and 8).
No changes made related to iOS platform. 

You probably need https://developers.google.com/web/tools/chrome-devtools/remote-debugging/webviews to inspect web3 object in JS console on remote device

Keep in mind that Service Worker API has been added to Android API 24 and higher. It means that DApp using Service Worker won't be loaded on Android phones with versions less than 7 https://developer.android.com/reference/android/webkit/ServiceWorkerController.html

status: ready


<blockquote><div><strong><a href="https://s3-us-west-1.amazonaws.com/turnsy/dapp/www/index.html">Ionic App</a></strong></div></blockquote>
<blockquote><img src="https://developers.google.com/web/images/social-webfu-16x9.png" width="48" align="right"><div>Google Developers</div><div><strong><a href="https://developers.google.com/web/tools/chrome-devtools/remote-debugging/webviews">Remote Debugging WebViews  |  Tools for Web Developers
       |  Google Developers</a></strong></div><div>Debug WebViews in your native Android apps using Chrome Developer Tools.</div></blockquote>
